library/kani/src/lib.rs

```diff
@@ -183,6 +183,39 @@ pub fn any_modifies<T>() -> T {
     unreachable!()
 }
 
+/// Recieves a reference to a pointer-like object and assigns kani::any_modifies to that object.
+/// Only for use within function contracts and will not be replaced if the recursive or function stub
+/// replace contracts are not used.
+#[rustc_diagnostic_item = "KaniHavoc"]
+#[inline(never)]
+#[doc(hidden)]
+pub fn havoc<T: ?Sized>(_pointer: &T) {
```

Comment:
I would suggest that we rename these `havoc` functions to follow the `any` convention. Maybe `write_any` since we are writing an arbitrary value into a memory location.